packages/qdrant-loader/src/qdrant_loader/core/state/session.py

```
@@ -5,23 +5,59 @@
+                echo=False,
+            )
+    else:
+        # asyncpg is an optional dependency (the `postgres` extra), imported
+        # lazily only on the Postgres path so SQLite installs don't need it.
+        try:
+            import asyncpg  # noqa: F401
+        except ModuleNotFoundError as exc:
+            raise RuntimeError(
+                "PostgreSQL state backend requires the 'postgres' extra: "
+                "install with `pip install qdrant-loader[postgres]`."
+            ) from exc
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)